### PR TITLE
Update terms of all documents

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib/jest-matchers/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type { ZodTypeAny } from 'zod';
```
Comment (paulpilone) [inline lib/jest-matchers/index.ts]: These changes are also in #126

File: lib/jest-matchers/jest.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import type { ZodTypeAny } from 'zod';
```
Comment (paulpilone) [inline lib/jest-matchers/jest.d.ts]: These changes are also in #126

File: lib/open-search-client.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 /* eslint-disable no-underscore-dangle */
```
Comment (paulpilone) [inline lib/open-search-client.ts:1]: These changes are also in #126

File: lib/open-search-client.ts [hunk 4]
```diff
@@ -380,3 +384,68 @@ export const deleteIndex = async (
...
+    while (hits.length > 0 && ctr < 2) {
+      await pMap(hits, f, pMapOptions);
+
+      const nextScrollResponse = scrollResponseSchema.parse(
+        await nextScroll(esUrl, scrollId)
+      );
+
+      hits = nextScrollResponse.hits.hits;
+
+      ctr += 1;
```
Comment (paulpilone) [inline lib/open-search-client.ts]: So at most we only loop through hits twice? What if we have more searchable documents than that?

File: neptune-bulk-loader/update-document-terms.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+});
+
+const updateDocumentTerms = async (
+  esUrl: string,
+  index: string,
+  rawHit: unknown
+): Promise<void> => {
+  const hit = hitSchema.parse(rawHit);
+
+  if (hit._source._bitstreamText === undefined) return;
```
Comment (paulpilone) [inline neptune-bulk-loader/update-document-terms.ts]: We still want to percolate a document even if it doesn't have text from a PDF. We just send the title.

File: lib/open-search-client.ts (hunk 4 above)
Comment (ghost) [reply lib/open-search-client.ts]: That was for testing and should have been removed. Fixed in 0cb0177

File: neptune-bulk-loader/update-document-terms.ts (hunk 5 above)
Comment (ghost) [reply neptune-bulk-loader/update-document-terms.ts]: Updated in dbe9d70

**Review by paulpilone — APPROVED** [review]

**Pull request merged**